regolith/helpers/l_contactshelper.py

```diff
@@ -0,0 +1,136 @@
+    subpi.add_argument("-no", "--notes", help='finds contacts based off notes or miscellaneous information about the contact', nargs = '+')
+    return subpi
+
+def is_current(thing, now=None):
+    """
+    given a thing with dates, returns true if the thing is current
+    looks for begin_ and end_ daty things (date, year, month, day), or just
+    the daty things themselves. e.g., begin_date, end_month, month, and so on.
+    Parameters
+    ----------
+    thing: dict
+      the thing that we want to know whether or not it is current
+    now: datetime.date object
+      a date for now.  If it is None it uses the current date.  Default is None
+    Returns
+    -------
+    True if the thing is current and false otherwise
+    """
+    if not now:
+        now = datetime.date.today()
+    dates = thing
+    current = False
+    try:
+        if dates.get("begin_date") <= now <= dates.get("end_date", datetime.date(5000, 12, 31)):
+            current = True
+    except:
+        raise RuntimeError(f"Cannot find begin_date in document:\n {thing}")
+    return current
+
+class ContactsListerHelper(SoutHelperBase):
+    """Helper for listing upcoming (and past) projectum milestones.
+
+       Projecta are small bite-sized project quanta that typically will result in
+       one manuscript.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        try:
+            if not rc.database:
+                rc.database = rc.databases[0]["name"]
+        except:
+            pass
+        colls = [
+            sorted(
+                all_docs_from_collection(rc.client, collname), key=_id_key
+            )
+            for collname in self.needed_dbs
+        ]
+        for db, coll in zip(self.needed_dbs, colls):
+            gtx[db] = coll
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+
+    def sout(self):
+        rc = self.rc
+        contacts = []
+        if rc.date:
+            temp_dat = date_parser.parse(rc.date).date()
+        for contact in self.gtx["contacts"]:
+            if rc.name:
+                list_add = False
+                for nam in rc.name:
+                     if nam.casefold() in contact.get('name').casefold():
+                        contacts.append(contact)
+                        list_add = True
+                if list_add:
+                    continue
+            if rc.inst and (rc.inst.casefold() in contact.get('institution').casefold()):
```

Comment:
I think I want AND logic for the institution and dates.  I think you are using OR logic.  Here is a UC:
```
1) Simon remembers he met the person around August 2019
2) Simon remembers she was from arizona state or u arizona
3) Simon types regolith helper l_contacts -i ariz -d 2019-08-15
4) l_contacts returns a very short list of possibilities, not everyone he ever met from arizona and everyone he met from anywhere in that date period
```
if you see what I mean.